lib/booking.rb

```diff
@@ -0,0 +1,81 @@
+require 'pry'
+require 'time'
+require 'date'
+
+module Hotel
+  class BookingSystem
+    attr_reader :rooms, :reservations
+
+    def initialize
+      # times loop to create an array of 20 rooms
+      @rooms = []
+      20.times do |room_num|
+        room_num += 1
+        @rooms << Room.new(room_num)
+      end
+      @reservations = []
+    end
+
+    # method to find available rooms and return an array of available room
+    def find_rooms_available(start_date, end_date)
+
+      rooms_available = []
+
+      reserv_range = DateRange.new(start_date, end_date)
+      x = reserv_range.get_date_range
+
+      @rooms.each do |room|
+        if room.room_reservations.empty?
+          rooms_available << room
+        elsif
+          room.room_reservations.each do |reservation|
```

Comment:
This is an example of tight coupling between `BookingSystem` and `Room`. This code requires the booking system to know how the internals of `Room` work, particularly how it stores reservations. A better approach would be to call a method, something like `Room#is_available?(date_range)`, and let `Room` handle all this logic.